src/signer.spec.ts

```diff
@@ -300,6 +305,52 @@ describe('Signer', () => {
     });
   });
 
+  describe('Should init Signer without owner and notify error regarding owner not set', () => {
+    let signer: Signer;
+    let requestId: string;
+    let postMessageMock: MockInstance;
+    const testOrigin = 'https://test.com';
+
+    beforeEach(() => {
+      requestId = crypto.randomUUID();
+      // Initialize the signer without setting an owner
+      signer = Signer.init(signerOptions);
+      postMessageMock = vi.fn();
+      vi.stubGlobal('opener', {postMessage: postMessageMock});
+    });
+
+    afterEach(() => {
+      signer.disconnect();
+      vi.clearAllMocks();
+      vi.restoreAllMocks();
+    });
+
+    it('should emit NOT_INITIALIZED error when a status message is received and no owner is set', async () => {
```

Comment:
Isn't this test actually incorrect? Status and supported standard should be emitted even if the signer is not initialized. Can you improve the code and add some tests in that direction?